### PR TITLE
Imran/lp 1042

### PR DESCRIPTION
**Pull request by emraan004**

### CODE REVIEW

File: lms/djangoapps/student_dashboard/views.py [hunk 1]
```diff
@@ -16,6 +17,26 @@
...
+        :param card_id:
+        :return:
+        """
+    utc = pytz.UTC
+
+    other_children_ids = [
+        crs.course_key for crs in CourseRerunState.objects.filter(
+            source_course_key=card_id
+        )
+    ]
```
Comment (muhammadnadeem) [inline lms/djangoapps/student_dashboard/views.py]: Please add action='rerun', state='succeeded' in filter

File: lms/djangoapps/student_dashboard/views.py [hunk 2]
```diff
@@ -16,6 +17,26 @@
...
+    future_courses = CourseOverview.objects.filter(
+        id__in=other_children_ids,
+        end__gt=datetime.utcnow().replace(tzinfo=utc))
```
Comment (muhammadnadeem) [inline lms/djangoapps/student_dashboard/views.py]: Can we get only one ongoing run to implement our logic?

File: lms/djangoapps/student_dashboard/views.py [hunk 3]
```diff
@@ -44,16 +65,29 @@ def get_enrolled_past_courses(course_enrollments):
     """
     # TODO move this function out of core code
     enrolled, past = [], []
+    past_course_cards = {}
 
     card_list = get_course_cards_list()
+
     for course in course_enrollments:
         course_card = get_related_card(course.course_overview)
```
Comment (muhammadnadeem) [inline lms/djangoapps/student_dashboard/views.py]: please using card list to avoid this line.

File: lms/djangoapps/student_dashboard/views.py [hunk 4]
```diff
@@ -44,16 +65,29 @@ def get_enrolled_past_courses(course_enrollments):
...
         if course_card in card_list:
             if course.course_overview.has_ended():
-                past.append(course)
+                if course_card.id not in past_course_cards:
+                    past_course_cards[course_card.id] = []
+
+                have_future_course = get_future_courses(course_card.id)
```
Comment (muhammadnadeem) [inline lms/djangoapps/student_dashboard/views.py]: please make a list before for loop and use that list inside loop to validate checks

File: lms/djangoapps/student_dashboard/views.py [hunk 5]
```diff
@@ -44,16 +65,29 @@ def get_enrolled_past_courses(course_enrollments):
...
+                if not have_future_course:
+                    course.course_overview.message = "No classes scheduled at the moment"
+
+                past_course_cards[course_card.id].append(course)
             else:
                 enrolled.append(course)
 
+    for card, courses in past_course_cards.items():
+        sorted_courses = sorted(courses, key=lambda course: course.course_overview.start)
+        past.append(sorted_courses[-1])
```
Comment (muhammadnadeem) [inline lms/djangoapps/student_dashboard/views.py]: please handle index error

Comment (emraan004) [reply lms/djangoapps/student_dashboard/views.py]: Loop won't execute if there are not items in it. hence it won't raise index out of bound error.

**Pull request merged**